cmd/longtail/main.go

```diff
@@ -716,17 +719,123 @@ func upSyncVersion(
 		timeStats = append(timeStats, timeStat{"Write version store index", writeVersionLocalStoreIndexTime})
 	}
 
+	if getConfigPath != nil && len(*getConfigPath) > 0 {
+		writeGetConfigStartTime := time.Now()
+
+		v := viper.New()
+		v.SetConfigType("json")
+		v.Set("storage-uri", blobStoreURI)
+		v.Set("source-path", targetFilePath)
+		if versionLocalStoreIndexPath != nil && len(*versionLocalStoreIndexPath) > 0 {
+			v.Set("version-local-store-index-path", *versionLocalStoreIndexPath)
+		}
+		tmpFile, err := ioutil.TempFile(os.TempDir(), "longtail-")
+		if err != nil {
+			return storeStats, timeStats, errors.Wrapf(err, "upSyncVersion: ioutil.TempFile() failed")
+		}
+		tmpFilePath := tmpFile.Name()
+		tmpFile.Close()
+		fmt.Printf("tmp file: %s", tmpFilePath)
+		err = v.WriteConfigAs(tmpFilePath)
+		if err != nil {
+			return storeStats, timeStats, errors.Wrapf(err, "upSyncVersion: v.WriteConfigAs() failed")
+		}
+
+		bytes, err := ioutil.ReadFile(tmpFilePath)
+		if err != nil {
+			return storeStats, timeStats, errors.Wrapf(err, "upSyncVersion: ioutil.ReadFile(%s) failed", tmpFilePath)
+		}
+		os.Remove(tmpFilePath)
+
+		err = longtailstorelib.WriteToURI(*getConfigPath, bytes)
+		if err != nil {
+			return storeStats, timeStats, errors.Wrapf(err, "upSyncVersion: longtailstorelib.WriteToURI(%s) failed", *getConfigPath)
+		}
+
+		writeGetConfigTime := time.Since(writeGetConfigStartTime)
+		timeStats = append(timeStats, timeStat{"Write get config", writeGetConfigTime})
+	}
+
 	return storeStats, timeStats, nil
 }
 
+func getVersion(
+	getConfigPath string,
+	targetFolderPath *string,
+	targetIndexPath *string,
+	localCachePath *string,
+	retainPermissions bool,
+	validate bool,
+	includeFilterRegEx *string,
+	excludeFilterRegEx *string) ([]storeStat, []timeStat, error) {
+
+	storeStats := []storeStat{}
+	timeStats := []timeStat{}
+
+	readGetConfigStartTime := time.Now()
+
+	vbuffer, err := longtailstorelib.ReadFromURI(getConfigPath)
+	if err != nil {
+		return storeStats, timeStats, errors.Wrapf(err, "getVersion: longtailstorelib.ReadFromURI() failed")
+	}
+
+	v := viper.New()
+	v.SetConfigType("json")
+	err = v.ReadConfig(bytes.NewBuffer(vbuffer))
+	if err != nil {
+		return storeStats, timeStats, errors.Wrapf(err, "getVersion: v.ReadConfig() failed")
+	}
+
+	blobStoreURI := v.GetString("storage-uri")
+	if blobStoreURI == "" {
+		return storeStats, timeStats, fmt.Errorf("getVersion: missing storage-uri in get-config")
+	}
+	sourceFilePath := v.GetString("source-path")
+	if sourceFilePath == "" {
+		return storeStats, timeStats, fmt.Errorf("getVersion: missing source-path in get-config")
+	}
+	var versionLocalStoreIndexPath string
+	if v.IsSet("version-local-store-index-path") {
+		versionLocalStoreIndexPath = v.GetString("version-local-store-index-path")
+	}
+
+	readGetConfigTime := time.Since(readGetConfigStartTime)
+	timeStats = append(timeStats, timeStat{"Read get config", readGetConfigTime})
+
+	resolvedTargetFolderPath := ""
```

Comment:
Move this to downSyncVersion and make `targetPath` optional for `downSync` as well